qgis_deployment_toolbelt/utils/user_groups.py

```diff
@@ -0,0 +1,284 @@
+            is used. Defaults to None.
+
+    Raises:
+        NotImplementedError: if operating system is not supported.
+
+    Returns:
+        list[str]: sorted list of unique groups names
+    """
+    if user_name is None:
+        user_name = getuser()
+
+    if not is_computer_attached_to_a_domain():
+        logger.debug(
+            "Computer is not attached to a domain so retrieving user's domain groups "
+            "is meaningless. Returning empty list."
+        )
+        return []
+
+    if opersys.lower() in ("darwin", "linux"):
+        # TODO: find a way to retrive user's domain groups on Linux and MacOS (probably
+        #  using pure ldap)
+        return []
+    elif opersys.lower() in ("win32", "windows"):
+        if pyad is not None:
+            user_obj = pyad.aduser.ADUser.from_cn(getuser())
+            return sorted(set(user_obj.get_attribute("memberOf")))
+        return []
+    else:
+        raise NotImplementedError(f"Unsupported operating system: {opersys}")
+
+
+def _win32_is_computer_in_domain() -> bool:
+    """Check if the computer is joined to a domain or a workgroup using WMI.
+
+    Returns:
+        bool: True if the computer is joined to a domain or workgroup, False otherwise.
+    """
+    try:
+        # Connect to the WMI service
+        wmi = win32com.client.GetObject("winmgmts://./root/cimv2")
+
+        # Execute a query to retrieve information about the domain
+        query = "SELECT * FROM Win32_ComputerSystem"
+        result = wmi.ExecQuery(query)
+
+        # Check if the domain attribute is not None
+        for item in result:
+            if item.Domain and item.Domain.lower() != "workgroup":
+                return True
+
+    except Exception as err:
+        logger.error(f"Getting domain from WMI through win32com failed. Trace: {err}")
+        raise err
+
+    return False
+
+
+@lru_cache
+def is_computer_attached_to_a_domain() -> bool:
+    """Determine if the computer is attached to a domain or not.
+
+    On Linux and MacOS, it always return False.
+    On Windows, it tries to use wmi (Windows Management Instrumentation), Active
+        Directory (through pyad) or subprocessed powershell as final fallback.
+
+    Raises:
+        NotImplementedError: if operating system is not supported.
+
+    Returns:
+        bool: True if the computer is attached to a domain.
+    """
+    if opersys.lower() in ("darwin", "linux"):
+        # TODO: find a way to determine if a computer is attached to a domain on Linux and MacOS
+        return False
+    elif opersys.lower() in ("win32", "windows"):
+        if wmi is not None:
+            if hasattr(wmi, "WMI"):
+                try:
+                    wmi_os = wmi.WMI().Win32_ComputerSystem()[0]
+                    logger.info(
+                        "Based on WMI, is the computer attached to a domain: "
+                        f"{wmi_os.PartOfDomain}"
+                    )
+                    return wmi_os.PartOfDomain
```

Comment:
Maybe we could add a function `_wmi_is_computer_in_domain`  similar to function `_win32_is_computer_in_domain`